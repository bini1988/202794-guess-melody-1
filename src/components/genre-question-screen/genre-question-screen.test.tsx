import React from "react";
import renderer from "react-test-renderer";
import {TrackGenres} from "../../types.d";
import GenreQuestionScreen from "./genre-question-screen";

it(`GenreQuestionScreen correctly renders default markup`, () => {
  const answersMock = [
    {src: ``, genre: TrackGenres.Rock},
    {src: ``, genre: TrackGenres.Jazz}
  ];

  const tree = renderer
    .create(
        <GenreQuestionScreen
          index={0}
          genre={TrackGenres.Rock}
          answers={answersMock}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
