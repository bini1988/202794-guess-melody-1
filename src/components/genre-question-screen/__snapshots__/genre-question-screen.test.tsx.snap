// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen correctly renders default markup 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите rock треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio>
          <source
            src=""
            type="audio/ogg; codecs=vorbis"
          />
        </audio>
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-0"
          name="answer"
          type="checkbox"
          value="rock"
        />
        <label
          className="game__check"
          htmlFor="answer-0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio>
          <source
            src=""
            type="audio/ogg; codecs=vorbis"
          />
        </audio>
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-1"
          name="answer"
          type="checkbox"
          value="jazz"
        />
        <label
          className="game__check"
          htmlFor="answer-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
